cpp/src/arrow/util/hashing.h

```diff
@@ -843,6 +843,14 @@ class BinaryMemoTable : public MemoTable {
     }
   }
 
+  // Visit the stored value at a specific index in insertion order.
+  // The visitor function should have the signature `void(std::string_view)`
```

Comment:
`const std::string_view&` doesn't add a lot of avlues in most contexts is there  a reason this signature needs to be supported)?

Reply:
I suppose it's costless to do so anyway.

Reply:
I agree that `const std::string_view&` is not recommended but I think it is better to match the function `VisitValues` above.